lib/irb/completion.rb

```diff
@@ -11,7 +11,29 @@
 
 module IRB
   module InputCompletor # :nodoc:
+    using Module.new {
+      refine ::Binding do
+        def eval_methods
+          eval("::Kernel.instance_method(:methods).bind(self).call")
```

Comment:
Instead of wrapping the entire chain inside `eval`, can we only wrap the necessary part?

```rb
        def eval_methods
          ::Kernel.instance_method(:methods).bind(eval("self")).call
        end

        def eval_private_methods
          ::Kernel.instance_method(:private_methods).bind(eval("self")).call
        end

        def eval_instance_variables
          ::Kernel.instance_method(:instance_variables).bind(eval("self")).call
        end

        def eval_global_variables
          ::Kernel.instance_method(:global_variables).bind(eval("self")).call
        end

        def eval_class_constants
          ::Module.instance_method(:constants).bind(eval("self.class")).call
        end
```

Reply:
Thanks comments :)
I fixed it.
https://github.com/ruby/irb/pull/369/commits/78ef990c1ce5a4b82b7dbce8dc72277122579011